### DISCUSSION TITLE
substructure search is too slow for Salts and explicit Hydrogen if running with QMOL query

### DISCUSSION THREAD

**Discussion by epbpo** [rdkit/rdkit#7670] [Development]
[Rdkit issue Statement.pdf](https://github.com/user-attachments/files/16412139/Rdkit.issue.Statement.pdf)

I have mentioned my issue in this document .Kindly assist me as I am struggling from past 2 months and not sure about root cause.
We have gist index on mol_data (RDKIT object)

**Comment by kienerj** [comment]
Someone more involved will have to add to my answer. I have below remarks:

- AFAIK only "normal" substructure searches are indexed, anything using QueryMol (=SMARTS) is not
- Maybe check out the [AdjustQueryProperties](https://www.rdkit.org/docs/source/rdkit.Chem.rdmolops.html#rdkit.Chem.rdmolops.AdjustQueryParameters) options which might help with your specific problem

I have a system at hand that uses rdkit cartridge and I don't see the mentioned Problems, both the charge and the explicit Hs seems to be respected. The processing of the search query matters as well and you likely need to take care that the explicit Hs aren't stripped when creating the rdkit molecule.

**Reply by epbpo** [reply]
![image](https://github.com/user-attachments/assets/8a03f544-a6bd-4550-a732-c5cf57b64f0a)
thanks for you response.
 We ran query analyzer and found this.  We had implemented gist index. Could you take a look and suggest me. It will be really helpful
